src/main/java/felix/utils/Parser.java

```diff
@@ -0,0 +1,89 @@
+     *
+     * @return Command.
+     * @throws FelixException If command is unrecognised or command formatting is wrong.
+     */
+    public Command getCommand(String line) throws FelixException {
+        // separate first word from rest of words
+        String[] words = line.split(" ", 2);
+        switch (words[0]) {
+        case "bye":
+            return new ExitCommand();
+        case "list":
+            return new ListCommand();
+        case "mark":
+            try {
+                return new MarkCommand(Integer.parseInt(words[1]) - 1);
+            } catch (NumberFormatException err) {
+                throw new FelixException("Enter a number after \"mark\"");
+            }
+        case "unmark":
+            try {
+                return new UnmarkCommand(Integer.parseInt(words[1]) - 1);
+            } catch (NumberFormatException err) {
+                throw new FelixException("Enter a number after \"unmark\"");
+            }
+        case "delete":
+            try {
+                return new DeleteCommand(Integer.parseInt(words[1]) - 1);
+            } catch (NumberFormatException err) {
+                throw new FelixException("Enter a number after \"delete\"");
+            }
+        case "todo":
+        case "deadline":
+        case "event":
+            return new AddTaskCommand(this.generateTask(words));
+        case "find":
+            try {
+                return new FindCommand(words[1]);
+            } catch (IndexOutOfBoundsException err) {
+                throw new FelixException("keyword cannot be empty");
+            }
+        default:
+            throw new FelixException("Unrecognised command");
+        }
+    }
+
+    private Task generateTask(String[] words) throws FelixException {
+        Task task;
+        if (words[0].equals("todo")) {
+            try {
+                task = new ToDo(words[1]);
+            } catch (IndexOutOfBoundsException e) {
+                throw new FelixException("The description of a todo cannot be empty");
+            }
+        } else if (words[0].equals("deadline")) {
+            try {
+                String[] remainder = words[1].split(" /by ");
+                task = new Deadline(remainder[0], remainder[1]);
+            } catch (IndexOutOfBoundsException e) {
+                throw new FelixException("Command does not follow this format: deadline {description} by {end_datetime}");
+            } catch (DateTimeParseException e) {
+                throw new FelixException("datetime for deadline is not in the format \"yyyy-MM-dd HHmm\"");
+            }
+        } else {
+            try {
+                String[] remainder = words[1].split(" /from | /to ");
+                task = new Event(remainder[0], remainder[1], remainder[2]);
+            } catch (IndexOutOfBoundsException e) {
+                throw new FelixException("Command does not follow this format: event {description} /from {start_datetime} /to {end_datetime}");
```

Comment:
Should this line be wrapped? I noticed the same issue in other places too.

Reply:
Thank you for the feedback! Just to clarify, what do you mean by wrapping the line?

Reply:
Hi, the course coding standard states that lines should not pass the soft limit of 110 lines, and a hard limit of 120 lines. The oracle coding standard states 80 lines. This line is really long so it could be changed to look (for e.g.) like
```java
throw new FelixException("Command does not follow this format: event {description} "
        + "/from {start_datetime} /to {end_datetime}");
```